tests/testthat/test-plotly-colorscale.R

```diff
@@ -0,0 +1,72 @@
+context("colorscales")
+
+
+test_that("Can specify marker.colorscale", {
+  p <- plot_ly(
+    x = c(-9, -6, -5, -3, -1), 
+    y = c(0, 1, 4, 5, 7), 
+    marker = list(
+      color = 1:5, 
+      colorscale='Rainbow', 
+      showscale = TRUE
+    )
+  ) 
+  l <- expect_doppelganger_built(p, "marker.colorscale")
+})
+
+test_that("Can specify contour colorscale", {
+  p <- plot_ly(
+    x = c(-9, -6, -5, -3, -1), 
+    y = c(0, 1, 4, 5, 7), 
+    z = matrix(c(10, 10.625, 12.5, 15.625, 20, 5.625, 6.25, 8.125, 11.25, 15.625, 2.5, 3.125, 5, 8.125, 12.5, 0.625, 1.25, 3.125,
+                 6.25, 10.625, 0, 0.625, 2.5, 5.625, 10), nrow = 5, ncol = 5), 
+    type = "contour",
+    colorscale = 'Rainbow'
+  )
+  l <- expect_doppelganger_built(p, "contour-colorscale")
+})
+
+test_that("Can provide a color interpolation function", {
+  p <- plot_ly(x = 1:10, y = 1:10, color = 1:10, colors = scales::colour_ramp(c("red", "green")))
+  l <- expect_doppelganger_built(p, "colorRamp")
+})
+
+test_that("Can specify contour colorscale", {
+  
+  plot_colorscale <- function(colorscale) {
+    plot_ly(
+      x = 1:10, 
+      y = 1:10, 
+      marker = list(
+        color = 1:10, 
+        colorscale = colorscale, 
+        showscale = TRUE
+      )
+    )
+  }
```

Comment:
Another test with `NA`s  for `marker.color`/`marker.colorscale`

Reply:
It seems plotly.js simply ignores missing values in the colorscale, but if you have missing values in `color`/`z`, it will color those in gray.

```r
plot_ly(
  x = 1:10, 
  y = 1:10, 
  marker = list(
    color = c(1:9, NA), 
    showscale = TRUE
  )
)
```

<img width="1094" alt="Screen Shot 2019-03-12 at 6 22 34 PM" src="https://user-images.githubusercontent.com/1365941/54242781-d999fc80-44f3-11e9-8973-e35dc5d080b6.png">

As it turns out, there is currently no way to specify the 'NA color', but it's on plotly.js' radar